tests/test_script.py

```diff
@@ -104,6 +108,152 @@ def test_export_slice_housing_multiple():
+    valid_yaml = """
+
+- module: keras.engine.training
+  annotations:
+    - criteria:
+        class_method_name: compile
+        class_instance: Model
+      side_effects:
+        - mutated_value:
+            self_ref: SELF_REF
+    """
+    with tempfile.NamedTemporaryFile(suffix=".not-yaml") as f:
+        f.write(valid_yaml.encode())
+        f.flush()
+        with pytest.raises(subprocess.CalledProcessError):
+            subprocess.check_call(["lineapy", "annotate", "add", f.name])
+
+    proc = subprocess.run(
+        ["lineapy", "annotate", "list"], check=True, capture_output=True
+    )
+    assert len(proc.stdout) == 0
+
+
+@pytest.mark.slow
+def test_annotate_add_invalid_yaml(annotations_folder):
+    invalid_yaml = """
+
+- module: keras.engine.training
+  error in yaml
+  annotations:
+    - criteria:
+        class_method_name: compile
+        class_instance: Model
+      side_effects:
+        - mutated_value:
+            self_ref: SELF_REF
+    """
+    with tempfile.NamedTemporaryFile(suffix=".yaml") as f:
+        f.write(invalid_yaml.encode())
+        f.flush()
+        with pytest.raises(subprocess.CalledProcessError):
+            subprocess.check_call(["lineapy", "annotate", "add", f.name])
+
+    proc = subprocess.run(
+        ["lineapy", "annotate", "list"], check=True, capture_output=True
+    )
+    assert len(proc.stdout) == 0
+
+
+@pytest.mark.slow
+def test_annotate_add_valid_yaml(annotations_folder):
+    valid_yaml = """
+
+- module: keras.engine.training
+  annotations:
+    - criteria:
+        class_method_name: compile
+        class_instance: Model
+      side_effects:
+        - mutated_value:
+            self_ref: SELF_REF
+    """
+    with tempfile.NamedTemporaryFile(suffix=".yaml") as f:
+        f.write(valid_yaml.encode())
+        f.flush()
+        subprocess.check_call(["lineapy", "annotate", "add", f.name])
+
+    proc = subprocess.run(
+        ["lineapy", "annotate", "list"], check=True, capture_output=True
+    )
+    assert len(str(proc.stdout).split("\n")) == 1
+
+
+def test_annotate_delete_invalid_path(tmp_path, annotations_folder):
+    invalid_path = tmp_path / "nonexistent.yaml"
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.check_call(
+            ["lineapy", "annotate", "delete", str(invalid_path)]
+        )
+
+
+def test_delete_existing_source(annotations_folder):
+    valid_yaml = """
+
+- module: keras.engine.training
+  annotations:
+    - criteria:
+        class_method_name: compile
+        class_instance: Model
+      side_effects:
+        - mutated_value:
+            self_ref: SELF_REF
+    """
+    with tempfile.NamedTemporaryFile(suffix=".yaml") as f:
+        f.write(valid_yaml.encode())
+        f.flush()
+        subprocess.check_call(["lineapy", "annotate", "add", f.name])
+
+    subprocess.check_call(
+        ["lineapy", "annotate", "delete", os.path.basename(f.name)]
+    )
```

Comment:
this should be inside the context. f might be defined outside the context i think but the actual file will be deleted as soon as you are out.

Reply:
resolving this as this wasnt the issue. the tempfilename had chars that got removed. running slugify on the basename should fix it.